website/docs/language/resources/ephemeral/write-only.mdx

```diff
@@ -1,44 +1,128 @@
+
+<CodeBlockConfig highlight="7">
+
+```hcl
+resource "aws_db_instance" "test" {
+  instance_class      = "db.t5.micro"
+  allocated_storage   = "5"
+  engine              = "postgres"
+  username            = "admin"
+  skip_final_snapshot = true
+  password_wo         = <ephemeral or non-ephemeral value>
+  password_wo_version = 1
+}
+```
+
+</CodeBlockConfig>
+
+Write-only arguments accept both ephemeral and non-ephemeral values. For example, you could also use a string as the value of a write-only argument:
+
+```hcl
+resource "aws_db_instance" "test" {
+  instance_class      = "db.t5.micro"
+  allocated_storage   = "5"
+  engine              = "postgres"
+  username            = "admin"
+  skip_final_snapshot = true
+  password_wo         = "my-password-here"
+  password_wo_version = 1
+}
+```
+
+However, we recommend using write-only arguments for passing ephemeral values to resources. For example, you can use an `ephemeral` resource to generate a random password and pass it to the `password_wo` write-only argument:
+
+<CodeBlockConfig highlight="11">
+
+```hcl
+ephemeral "random" "password" {
+  length = 16
+}
+
+resource "aws_db_instance" "test" {
+  instance_class      = "db.t5.micro"
+  allocated_storage   = "5"
+  engine              = "postgres"
+  username            = "admin"
+  skip_final_snapshot = true
+  password_wo         = ephemeral.random.password.value
+  password_wo_version = 1
+}
+```
 
--> **Public Beta**: Write-only arguments are in public beta and available in Terraform v1.11 and later.  Public beta features and APIs are subject to change.
+</CodeBlockConfig>
 
+During a Terraform operation, the provider uses the `password_wo` value to configure the database instance, and then Terraform discards that value without storing it in the plan or state file. 
 
-## Introduction
+## Update write-only arguments with versions
 
-Write-only arguments are only available during runtime, and Terraform omits them from state and plan files. On a new Terraform operation, a write-only argument always start as `null` before Terraform overwrites it with a new value from your configuration.
+Terraform does not store write-only arguments in state files, so Terraform has no way of knowing if a write-only argument value has changed. Because Terraform cannot track write-only argument values, it sends write-only arguments to the provider during every operation.
 
-Write-only arguments are unique among other ephemeral constructs in Terraform because you can assign both ephemeral and non-ephemeral data as the value of a write-only argument.
+Terraform also cannot create plan diffs for write-only arguments because it does not store them in plan files, so providers only use write-only arguments when creating a new resource.
```

Comment:
I think this sentence implies that providers cannot use write-only arguments to update a resource. 
```suggestion
Terraform also cannot create plan diffs for write-only arguments because it does not store them in plan files, so providers typically implement mechanisms such as version arguments to indicate a plan diff to Terraform.
```

Reply:
I was trying to explain what happens if you _don't_ add an accompanying version arg (because I know most of them are optional), but I like your suggestion of going straight to version args. How does the following feel for making your suggestion a bit shorter? Since the next paragraph explains the why behind version args:
```suggestion
Terraform also cannot create plan diffs for write-only arguments because it does not store those values in plan files. However, providers typically include version arguments alongside write-only arguments. Terraform stores version arguments in state, and can track if a version argument changes. 
```

Reply:
That sounds good to me!